migrate_metedata/migrator.py

```diff
@@ -14,32 +14,42 @@ class Migrator:
 
     def __init__(self, ckan_classic_api, new_metastore, options):
         self.ckan_classic_api = ckan_classic_api
-        self.new_metastore = new_metastore
-        self.config = options
+        self.metastore_client = store.create_metastore(new_metastore, **options)
 
     def migrate_dataset(self, pkg_name):
-        urlpath = self.ckan_classic_api + '/package_show?id=' + pkg_name
+        urlpath = self.create_url('/package_show?id=', pkg_name)
+        response = json.loads(urllib.urlopen(urlpath).read())
+        self.create_pkg(response, pkg_name)
 
+    def migrate_all_datasets(self):
+        urlpath = self.create_url('/package_list')
         response = json.loads(urllib.urlopen(urlpath).read())
-        pkg_dict = response['result']
+        self.iterate_pkgs(response)
 
-        package_id = pkg_dict["id"]
-        data_package_json = converter.dataset_to_datapackage(pkg_dict)
+    def create_url(self, suffix, pkg_name=''):
+        url = self.ckan_classic_api + suffix
+        url += pkg_name
+        return url
 
-        try:
-            metastore_client = store.create_metastore(self.new_metastore, **self.config)
-            package_info = metastore_client.create(package_id, data_package_json)
-        except Exception as e:
-            log.info("{} dataset is already exists".format(pkg_name))
+    def create_pkg(self, data, pkg_name):
+        pkg_dict = data.get('result')
 
-    def migrate_all_datasets(self):
-        urlpath = self.ckan_classic_api + '/package_list'
+        if pkg_dict:
+            data_package_json = converter.dataset_to_datapackage(pkg_dict)
+
+            try:
+                package_info = self.metastore_client.create(pkg_name, data_package_json)
+                return True
+            except store.exc.Conflict as e:
+                log.info("{} dataset already exists".format(pkg_name))
+                return False
+
+        return False
 
-        try:
-            response = json.loads(urllib.urlopen(urlpath).read())
-            pkgs_list = response['result']
-        except ValueError as e:
-            log.info("There is no dataset available in your CKAN")
+    def iterate_pkgs(self, data):
```

Comment:
There is some kind of unclear boundary here between the calling scope and this function. Why parse the JSON in the calling scope but fetch `result` out of it here? I think the boundary should be between what I'd call "envelope" and "data". Think how easier it would be to test this function if you just passed the extracted data to it without any "envelope" stuff like the fact that it is wrapped in `{"result": [...your data here...]}`? 

But really, if you analyze this code more deeply you will probably notice that there is not much here beyond a for loop: first, the `if not pkg_list` part is really not necessary. Why do you care to throw an exception if the list is empty? Just let it do nothing (iterate over an empty list). This is not an "error" really. If we get rid of that, and we get rid of extracting the `'result'`, all you are left with is a for loop. That can be easily merged into the calling scope. Eventually, you will see that once you start splitting flow control from logic and use some Pythonic tools such as generators, the iterator protocol etc. you end up with very little code that actually requires testing.